pilot/pkg/config/kube/crd/client.go

```diff
@@ -70,20 +60,81 @@ type IstioObjectList interface {
 
 // Client is a basic REST client for CRDs implementing config store
 type Client struct {
+	// Map of apiVersion to restClient.
+	clientset map[string]*restClient
+
+	// domainSuffix for the config metadata
+	domainSuffix string
+}
+
+type restClient struct {
+	apiVersion *schema.GroupVersion
+
+	// descriptor from the same apiVerion.
 	descriptor model.ConfigDescriptor
 
+	// types of the schema and objects in the descriptor.
+	types []*schemaType
+
 	// restconfig for REST type descriptors
 	restconfig *rest.Config
 
 	// dynamic REST client for accessing config CRDs
 	dynamic *rest.RESTClient
+}
 
-	// domainSuffix for the config metadata
-	domainSuffix string
+func apiVersion(schema *model.ProtoSchema) string {
+	return ResourceGroup(schema) + "/" + schema.Version
+}
+
+func apiVersionFromConfig(config *model.Config) string {
+	return config.Group + "/" + config.Version
+}
+
+func (cl *Client) newClientSet(descriptor model.ConfigDescriptor) error {
+	for _, typ := range descriptor {
+		s, exists := knownTypes[typ.Type]
+		if !exists {
+			return fmt.Errorf("missing known type for %q", typ.Type)
+		}
+
+		k := apiVersion(&typ)
+		if v, ok := cl.clientset[k]; !ok {
+			rc := new(restClient)
+			rc.apiVersion = &schema.GroupVersion{
+				ResourceGroup(&typ),
+				typ.Version,
+			}
```

Comment:
This could be simplified a bit, e.g.

```golang
rc, ok := cl.clientset[apiVersion(&typ)]
if !ok {
    // create a new client if one doesn't already exist
    rc = &restClient{apiVersion: schema.GroupVersion{ResourceGroup(&typ), typ.Version)}},
    cl.clientset[apiVersion(&typ)] = rc
}
rc.descriptor = append(rc.descriptor, typ)
rc.types = append(rc.types, &s)
```

Reply:
done